src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -427,15 +430,22 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
             ProgrammingExercise programmingExercise = programmingExerciseOpt.get();
             Course course = programmingExercise.getCourse();
 
-            Optional<Participation> assignmentParticipation = participationRepository.findByExerciseIdAndStudentIdWithLatestResult(programmingExercise.getId(), user.getId());
-            Set<Participation> participations = new HashSet<>();
+            Optional<StudentParticipation> assignmentParticipation = studentParticipationRepository.findByExerciseIdAndStudentIdWithLatestResult(programmingExercise.getId(),
+                    user.getId());
+            Set<StudentParticipation> participations = new HashSet<>();
             assignmentParticipation.ifPresent(participations::add);
             programmingExercise.setParticipations(participations);
 
             if (!authCheckService.isAtLeastInstructorForCourse(course, user)) {
                 return forbidden();
             }
 
+            // Avoid circular serialization issues with jackson.
+            if (programmingExercise.getTemplateParticipation() != null) {
```

Comment:
I committed this as a hotfix. If this is not done, there will be a circular serialization issue, even though `@JsonIgnoreProperties` is set on the programmingExercise in the Template/SolutionParticipation classes.